astroquery/esa/xmm_newton/core.py

```diff
@@ -297,9 +301,146 @@ def _parse_filename(self, filename):
+        ret = {}
+        if instrument == []:
+            instrument = _instrument
+        else:
+            for i in instrument:
+                if i not in _instrument:
+                    log.warning("Invalid instrument %s" % i)
+                    instrument.remove(i)
+        if path != "" and os.path.exists(path):
+            _path = path
+        try:
+            with tarfile.open(filename, "r") as tar:
+                for i in tar.getmembers():
+                    paths = os.path.split(i.name)
+                    fname = paths[1]
+                    paths = os.path.split(paths[0])
+                    if paths[1] != "pps":
+                        continue
+                    fname_info = self._parse_filename(fname)
+                    if fname_info["X"] != "P":
+                        continue
+                    if not fname_info["I"] in instrument:
+                        continue
+                    if not fname_info["T"] in _product_type:
+                        continue
+                    if int(fname_info["X-"], 16) != source_number:
+                        continue
+                    tar.extract(i, _path)
+                    key = fname_info["I"]
+                    value = os.path.abspath(os.path.join(_path, i.name))
+                    if fname_info["T"] == "BGSPEC":
+                        key = fname_info["I"] + "_bkg"
+                    elif fname_info["T"] == "SRCARF":
+                        key = fname_info["I"] + "_arf"
+                    else:
+                        with fits.open(value) as hdul:
+                            for ext in hdul:
+                                if ext.name != "SPECTRUM":
+                                    continue
+                                rmf_fname = ext.header["RESPFILE"]
+                                if fname_info["I"] == "M1" or \
+                                   fname_info["I"] == "M2":
+                                    inst = "MOS/" +\
+                                        str(ext.header["SPECDELT"]) + "eV/"
+                                elif fname_info["I"] == "PN":
+                                    inst = "PN/"
+                                    file_name, file_ext =\
+                                        os.path.splitext(rmf_fname)
+                                    rmf_fname = file_name +\
+                                        "_v18.0" + file_ext
+
+                                link = self._rmf_ftp + inst + rmf_fname
+
+                                if verbose:
+                                    log.info("rmf link is: %s" % link)
+
+                                response = self._request('GET', link)
+
+                                rsp_filename =\
+                                    os.path.join(_path,
+                                                 paths[0],
+                                                 paths[1],
+                                                 ext.header["RESPFILE"])
```

Comment:
These all should be in one line, I very much find it difficult to read this one word in a line formatting while the logical block is one single expression

Reply:
Done, thanks